src/handlers/serenity.rs

```diff
@@ -121,7 +136,14 @@ impl SerenityHandler {
                                     .name("query")
                                     .description("The media to play")
                                     .kind(CommandOptionType::String)
-                                    .required(true)
+                                    .required(false)
```

Comment:
Its been a bit, but wouldn't setting both of these arguments to required make a command with just one argument invalid?

Reply:
Ah, right, that's true! I was thinking whether or not it would be better to just have a different command, but I'm not sure if that's the best way forward.

Reply:
If we don't want people to even have the option of submitting a do-nothing play command they yeah, it should be moved to a separate /playfile command or something similar

Reply:
I agree, honestly I backported this from the code after I ported the entire thing to using poise which allows for typed parameters, so the attachment comes in typed already but needs to be a completely separate parameter. i.e.
```Rust
/// Play a song.
#[poise::command(slash_command, prefix_command, guild_only, aliases("p", "P"))]
pub async fn play(
    ctx: Context<'_>,
    #[description = "Play mode"] mode: Option<String>,
    #[description = "File to play."] file: Option<serenity::Attachment>,
    #[rest]
    #[description = "song link or search query."]
    query_or_url: Option<String>,
) -> Result<(), Error> {
```